extensions/applicationinsights-clickanalytics-js/src/handlers/AutoCaptureHandler.ts

```diff
@@ -0,0 +1,103 @@
+import { IAutoCaptureHandler, IPageActionOverrideValues, IClickAnalyticsConfiguration } from '../Interfaces/Datamodel'
+import { isRightClick, isLeftClick, isKeyboardEnter, isKeyboardSpace, isMiddleClick, isElementDnt } from '../common/Utils';
+import { ActionType } from '../Enums';
+import { ClickAnalyticsPlugin } from "../ClickAnalyticsPlugin";
+import { PageAction } from '../events/PageAction';
+
+const clickCaptureInputTypes = { BUTTON: true, CHECKBOX: true, RADIO: true, RESET: true, SUBMIT: true };
+
+export class AutoCaptureHandler implements IAutoCaptureHandler {
+    
+    /**
+     * @param analyticsPlugin - WebAnalytics plugin
+     * @param traceLogger - Trace logger to log to console.
+     */
+   constructor(protected _analyticsPlugin: ClickAnalyticsPlugin, protected _config: IClickAnalyticsConfiguration, protected _pageAction: PageAction,
+     protected _traceLogger: IDiagnosticLogger) {
+
+    }
+    // handle automatic event firing on user click
+    public click() {
+        let win = getWindow();
+        let doc = getDocument();
+        if (win) {
+            // IE9 onwards addEventListener is available, 'click' event captures mouse click. mousedown works on other browsers
+            const event = (navigator.appVersion.indexOf('MSIE') !== -1) ? 'click' : 'mousedown';
+            EventHelper.Attach(win, event , (evt:any) => { this._processClick(evt); });
+            EventHelper.Attach(win, 'keyup' , (evt:any) => { this._processClick(evt); });
+        } else if (doc && (doc as any).attachEvent) {
+            // IE8 and below doesn't have addEventListener so it will use attachEvent
+            // attaching to window does not work in IE8
+            (doc as any).attachEvent('onclick', (evt: Event) => { this._processClick(evt); });
+            (doc as any).attachEvent('keyup', (evt: Event) => { this._processClick(evt); });
+        }
+    }
+
+    /**
+     * API to create and send a populated PageAction event 
+     * @param element - DOM element
+     * @param overrideValues - PageAction overrides
+     * @param customProperties - Custom properties(Part C)
+     * @param isRightClick - Flag for mouse right clicks
+     */
+    private capturePageAction(element: Element, overrideValues?: IPageActionOverrideValues, customProperties?: { [name: string]: string | number | boolean | string[] | number[] | boolean[] | object }, isRightClick?: boolean): void {
+        const donotTrackTag = this._config.dataTags.customDataPrefix + this._config.dataTags.donotTrackDataTag;
+        if (!isElementDnt(element, donotTrackTag)) {
+            this._pageAction.capturePageAction(element, overrideValues, customProperties, isRightClick);
+        }
+
+    }
+
+    // Process click event
+    private _processClick(clickEvent: any) {
+        var clickCaptureElements = { A: true, BUTTON: true, AREA: true, INPUT: true };
+        let win = getWindow();
+        clickEvent = clickEvent || win.event; // IE 8 does not pass the event
```

Comment:
technically, based on the code above it's possible that win is not defined / null (lines 36/37) which would then cause the next line to fail if no clickEvent is passed. (I don't expect there to not be a clickEvent though)